CMakeLists.txt

```diff
@@ -1,5 +1,12 @@
 cmake_minimum_required(VERSION 3.15)
 
+# Skip generating proto files with Protobuf compiler and use prebuilt files instead
+# Will still build Protobuf as libprotobuf is required for linking
+option(WSL2_CROSS_COMPILE "Building on WSL2 for Windows" OFF)
```

Comment:
Could be a good idea to force OS check and bail out if target system is not windows or doesn't has installed mingw.

Reply:
As we are providing the toolchain file - with the expectant paths for the `mingw` toolchain - configuration will fail in this case:
```
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:10 (project):
  The CMAKE_C_COMPILER:

    /usr/bin/x86_64-w64-mingw3-gcc

  is not a full path to an existing compiler tool.

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.


CMake Error at CMakeLists.txt:10 (project):
  The CMAKE_CXX_COMPILER:

    /usr/bin/x86_64-w64-mingw3-g++

  is not a full path to an existing compiler tool.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.
  ```
  Here I've simply provided a bad path for the compilers as an example.
  
  But I do see value in building only if `LINUX`, as this feature has no value when building on Windows alone.